site/app/controllers/admin/ConfigurationController.php

```diff
@@ -71,55 +71,49 @@ public function viewConfiguration() {
-            $this->core->redirect($this->core->buildUrl(array('component' => 'admin',
-                                                              'page' => 'configuration',
-                                                              'action' => 'view')));
+        if(!isset($_POST['name'])) {
+            $response = array('status' => 'fail', 'message' => 'Name of config value not provided');
+            $this->core->getOutput()->renderJson($response);
+            return $response;
         }
+        $name = $_POST['name'];
 
-        foreach (array('default_hw_late_days', 'default_student_late_days') as $key) {
-            $_POST[$key] = (isset($_POST[$key])) ? intval($_POST[$key]) : 0;
+        if(!isset($_POST['entry'])) {
+            $response = array('status' => 'fail', 'message' => 'Name of config entry not provided');
+            $this->core->getOutput()->renderJson($response);
+            return $response;
         }
+        $entry = $_POST['entry'];
 
-        foreach (array('zero_rubric_grades', 'keep_previous_files', 'display_rainbow_grades_summary', 'display_custom_message', 'forum_enabled', 'regrade_enabled') as $key) {
-            $_POST[$key] = (isset($_POST[$key]) && $_POST[$key] == "true") ? true : false;
+        if($name === "course_name" && $entry === "") {
+            $response = array('status' => 'fail', 'message' => 'Course name cannot be blank');
+            $this->core->getOutput()->renderJson($response);
+            return $response;
         }
 
-        $save_array = array(
-            'course_details' => array(
-                'course_name'               => $_POST['course_name'],
-                'course_home_url'           => $_POST['course_home_url'],
-                'default_hw_late_days'      => $_POST['default_hw_late_days'],
-                'default_student_late_days' => $_POST['default_student_late_days'],
-                'zero_rubric_grades'        => $_POST['zero_rubric_grades'],
-                'upload_message'            => nl2br($_POST['upload_message']),
-                'keep_previous_files'       => $_POST['keep_previous_files'],
-                'display_rainbow_grades_summary' => $_POST['display_rainbow_grades_summary'],
-                'display_custom_message'    => $_POST['display_custom_message'],
-                'course_email'              => $_POST['course_email'],
-                'vcs_base_url'              => $_POST['vcs_base_url'],
-                'vcs_type'                  => $_POST['vcs_type'],
-                'forum_enabled'				=> $_POST['forum_enabled'],
-                'regrade_enabled'           => $_POST['regrade_enabled'],
-                'regrade_message'           => $_POST['regrade_message'],
-                'private_repository'        => $_POST['private_repository']
-            )
-        );
+        if(in_array($name, array('default_hw_late_days', 'default_student_late_days'))) {
+            $entry = $entry === "" ? intval($entry) : 0;
```

Comment:
you should check `ctype_digit`  on these parameters and `Output::renderJsonFail` if they aren't integers.  If you put a string in this field it will save a 0 and you wouldn't know that until you refresh the page.

Reply:
The boolean expression on 100 actually will only accept `$entry` if it is a blank string, which means this will always save a zero.

Reply:
whoops